purchase_manual_delivery/models/purchase_order.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2019 ForgeFlow S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    pending_to_receive = fields.Boolean(compute="_compute_pending_to_receive")
+    manual_delivery = fields.Boolean(
+        string="Manual delivery",
+        compute="_compute_manual_delivery",
+        help=(
+            "Stock transfers need to be created manually to receive this PO's products"
+        ),
+        readonly=False,
+        store=True,
+    )
+
+    @api.depends("company_id")
+    def _compute_manual_delivery(self):
+        """The manual delivery option is derived from the company of the order"""
+        for po in self:
+            po.manual_delivery = po.company_id.purchase_manual_delivery
+
+    def _compute_pending_to_receive(self):
+        for order in self:
+            order.pending_to_receive = True
+            if not any(order.mapped("order_line.pending_to_receive")):
+                order.pending_to_receive = False
+
+    def button_confirm_manual(self):
+        return super(
+            PurchaseOrder, self.with_context(manual_delivery=True)
+        ).button_confirm()
```

Comment:
Doesn't it seem like an obvious bug?
That it should be return `self.with_context(manual_delivery=True)
        ).button_confirm()`? Otherwise, you go back to purchase, and skip all overrides defined there. I'm not even sure that there's a good reason for this design (I frankly doubt it), but having all overrides in one case and not the other seems quite dangerous.

Reply:
If you mean, we should avoid creating a picking even when button_confirm is called directly in some way, I agree. I suggest we turn the mechanism around and make _create_picking do nothing for POs with manual_delivery=True *unless* a specific context key is present. We can then pass that context key in the wizard to force creating pickings for manualy delivery POs. Do you agree?

Reply:
I agree, that's the part where I suggested that I don't think it's a very good design (if the code of button_confirm_manual was directly put into an override of button_confirm that depends on a context key, you can still have 2 different buttons by putting the context in the attributes).

To make my point clearer, suppose you have module purchase_x that depends on purchase, and that it does an override of button_confirm to call `fx`.
Because `purchase_x` and `purchase_manual_delivery` have no relation to each other, we could either have `purchase_manual_delivery => purchase_x => purchase`, or `purchase_x => purchase_manual_delivery => purchase`.

If you call button confirm, in all cases you will call `fx` once. If you call button_confirm_manual, in one case you will call `fx` but not the other, depending on how the ORM loaded modules. If you decide that you want an easy solution and add an override of `button_confirm_manual` to also call `fx`, then you have the possibility to call `fx` twice, so you'd better be sure to make it idempotent.

Reply:
Ah, I see what you mean and you are right, that is a straightforward bug and can be fixed with 
```suggestion
        return self.with_context(manual_delivery=True).button_confirm()
```
Do you see additional value in the refactoring that I suggested?

Reply:
I sure do! Since a migration is the most appropriate time to do it, I  will make a PR with the simple fix for 17.0 and give you a review when you're done.

Reply:
I can do it, if you like.

Reply:
I need to rebase anyway.

Reply:
Just ping me when your refactoring is ready for review :-)

17.0 fix: https://github.com/OCA/purchase-workflow/pull/2573

Reply:
Here you go. I also added an additional fix for the fact that stock moves were still created when increasing quantities on order lines of a confirmed (manual delivery) order. Test failures are unrelated and fixed in https://github.com/OCA/purchase-workflow/pull/2576